src/chef-server-ctl/plugins/psql.rb

```diff
@@ -62,21 +61,26 @@
     db_config.merge!(JSON.parse(File.read("/etc/opscode-reporting/opscode-reporting-running.json")))
   end
 
-  seed=known_dbs[service_name]["hashseed"]
-  db_hash_key=known_dbs[service_name]["config_key"]
-  db_name=known_dbs[service_name]["dbname"]
+  seed = known_dbs[service_name]["hashseed"]
+  db_hash_key = known_dbs[service_name]["config_key"]
+  db_name = known_dbs[service_name]["dbname"]
 
   # undocumented, but available. Intended for use primarily for
   # gather-logs to be able to do its thing correctly.
   if (ARGV.include? "--as-admin")
     cfg = running_service_config('postgresql')
-    db_username=cfg['db_superuser']
-    db_password=credentials.get('postgresql', 'db_superuser_password')
+    db_username = cfg['db_connection_superuser'] || cfg['db_superuser']
+    db_password = credentials.get('postgresql', 'db_superuser_password')
   else
-    db_username=db_config[seed][db_hash_key]["sql_#{ro}user"]
+    if ARGV.include?(write_arg)
+      db_username = db_config[seed][db_hash_key]['sql_connection_user'] || db_config[seed][db_hash_key]['sql_user']
+    else
+      db_username = db_config[seed][db_hash_key]['sql_connection_user'] || db_config[seed][db_hash_key]['sql_ro_user']
```

Comment:
At this point the user is trying to connect to the db as a readonly user. Do not think we need to add db_config[seed][db_hash_key]['sql_connection_user'] string here.

Do we need to create another `sql_connection_user_ro` for this purpose?

Reply:
POSSIBLE OPTIONS:

1) Put instructions to the user in a source code comment advising how to invoke `psql` and under what conditions.

2) Create an `sql_ro_connection_user`.

3) Proceed with current modifications 'as-is.'

Reply:
I went a 4th way (a variation of #2 above).  If it needs a readonly user it uses the existing one (/w associated password).  If it needs an azure connection it adds the associated connection info into the connection string e.g. sql_ro_user@azure_ip.whatever.com.  If it doesn't need an azure connection, it doesn't add the associated garbage into the connection string.

Reply:
See test results on this somewhere below.